packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page.tsx

```diff
@@ -55,6 +62,11 @@ const Page: NextPageWithLayout = () => {
                   ...subgraph,
                   vertices: {
                     ...subgraph.vertices,
+                    /**
+                     * @todo - This is a problem, entity records should be immutable, there will be a new identifier
+                     *   for the updated entity. For places where we mutate elements, we should probably store them
+                     *   separately from the subgraph to allow for optimistic updates without being incorrect.
+                     */
```

Comment:
This is quite a big concern to me. I think we basically need to treat `Subgraph`s as immutable. They're a view of the graph, so it's a bad idea to go changing the internals of various entities and such. 

I think anything you're mutating locally should be stored separately in local state and you only update the graph with it when you get it back? I don't know how we can avoid that unless clients generate the IDs or can have some stand-in ID that can be used optimistically and then replaced when the request comes back.

I don't really want to completely rework the frontend code in this PR though, I think this would be a better use of someone else's time who is more familiar.

cc: @CiaranMn, @benwerner01

Reply:
This has changed a bit but still a concern to me:
https://github.com/hashintel/hash/blob/d91204a4e95303ce6d890f86438267532fe3761c/packages/hash/frontend/src/pages/%5Baccount-slug%5D/entities/%5Bentity-id%5D.page.tsx#L64-L84

Reply:
I agree, I think the entity that is being modified should be stored in state separately from the surrounding subgraph. 

(cc @yusufkinatas)